protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -395,22 +456,65 @@ impl BridgeContractCounterparty for MovementClient {
 		recipient: RecipientAddress<Self::Address>,
 		amount: Amount,
 	) -> BridgeContractCounterpartyResult<()> {
-		//@TODO properly return an error instead of unwrapping
+		
+		let test_noop_payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"noop_for_testing",
+			Vec::new(),
+			Vec::new(),
+		);
+
+		let _ = utils::send_and_confirm_aptos_transaction(
+			&self.rest_client,
+			self.signer.as_ref(),
+			test_noop_payload,
+		)
+		.await
+		.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
+
+		let test_u64_args = vec![
+			bcs::to_bytes(&initiator.0).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+			bcs::to_bytes(&bridge_transfer_id.0[..]).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+			bcs::to_bytes(&hash_lock.0[..]).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+			bcs::to_bytes(&time_lock.0).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+			bcs::to_bytes(&recipient.0.0).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+			bcs::to_bytes(&amount.0).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
+		];
```

Comment:
To reduce the repeating code, use an inline util fn, then bubble the err with `?`

```rust
let serialize = |value| {
    bcs::to_bytes(value).map_err(|_| BridgeContractCounterpartyError::SerializationError)
};

let test_u64_args = vec![
    serialize(&initiator.0)?,
    serialize(&bridge_transfer_id.0[..])?,
    serialize(&hash_lock.0[..])?,
    serialize(&time_lock.0)?,
    serialize(&recipient.0.0)?,
    serialize(&amount.0)?,
];
```

Reply:
Okay. That specific snippet is throwing errors for me; trying to debug.

Reply:
The closure was expecting values to all have the same type of `Vec<u8>`.

I solved it by creating two utility functions: one for `Vec<u8>` and one for `u64`.

Pushed update.